### PR TITLE
various StratCon fixes

### PR DESCRIPTION
**Pull request by NickAragua**
- Updated arrival altitude for aerospace fighter reinforcement modifiers (they were showing up on the ground)
- Standardized "recon" type scenarios to have the same criteria as Tukayyid scenarios 
- Various briefing text clarifications
- Store artillery offboard status between campaign saves
- On Integrated command, don't immediately and incorrectly undeploy lances after generating scenarios
- Adjusted manual force deployment rules (unlimited stacking on allied facilities, one per hex otherwise, no manual deployment under integrated command)
- Tighten up reinforcement restrictions (can't deploy units that can't actually deploy; can't double deploy units; can now only deploy fight lances or have to use a support point/VP)
- VTOLs no longer start on the ground

### CODE REVIEW

File: MekHQ/data/scenariomodifiers/FacilityHostileExtract.xml [hunk 1]
```diff
@@ -32,14 +32,15 @@
             <additionalDetails>
 		<additionalDetail>Extraction can be carried out by a unit with at least one free hand actuator, infantry or battle armor.</additionalDetail>
 		<additionalDetail>Extraction cannot be performed from destroyed buildings.</additionalDetail>
+		<additionalDetail>The extracting units may flee the battlefield, or the objective is automatically complete if routing the oppossing force.</additionalDetail>
```
Comment (Windchild292) [inline MekHQ/data/scenariomodifiers/FacilityHostileExtract.xml:35]:
```suggestion
		<additionalDetail>The extracting units may flee the battlefield, or the objective is automatically complete if routing the opposing force.</additionalDetail>
```

Comment (Windchild292) [reply MekHQ/data/scenariomodifiers/FacilityHostileExtract.xml:35]: automatically completed upon routing?

File: MekHQ/data/scenariotemplates/Destroy Grounded Dropship.xml [hunk 2]
```diff
@@ -169,8 +169,10 @@
                     <howMuch>2</howMuch>
                 </failureEffect>
             </failureEffects>
-            <additionalDetails/>
-            <description>Destroy, cripple or force to withdraw the following force(s) and unit(s):</description>
+            <additionalDetails>
+                <additionalDetail></additionalDetail>
+            </additionalDetails>
```
Comment (Windchild292) [inline MekHQ/data/scenariotemplates/Destroy Grounded Dropship.xml:174]: Why the addition of this?

File: MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java [hunk 3]
```diff
@@ -1139,10 +1132,67 @@ public static ReinforcementEligibilityType getReinforcementType(int forceID, Str
...
+        StratconFacility facility = track.getFacility(coords);
+        boolean alliedFacility = (facility != null) && (facility.getOwner() == ForceAlignment.Allied);
+        
+        return (!track.areAnyForceDeployedTo(coords) || alliedFacility) && nonCloakedOrNoscenario;
+    }
+    
+    /**
+     * Can the given force be manually deployed to the given coordinates on the given track
+     */
+    public static boolean canManuallyDeployForce(StratconCoords coords,
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java:1182]: Leaving it in, but unused, for now?

File: MekHQ/src/mekhq/gui/stratcon/StratconScenarioWizard.java [hunk 4]
```diff
@@ -418,7 +417,7 @@ private String buildForceCost(int forceID) {
         StringBuilder costBuilder = new StringBuilder();
         costBuilder.append("(");
         
-        switch(StratconRulesManager.getReinforcementType(forceID, currentTrackState, campaign)) {
+        switch(StratconRulesManager.getReinforcementType(forceID, currentTrackState, campaign, currentCampaignState)) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/stratcon/StratconScenarioWizard.java:420]:
```suggestion
        switch (StratconRulesManager.getReinforcementType(forceID, currentTrackState, campaign, currentCampaignState)) {
```

**Review by Windchild292 — APPROVED** [review]
Looks good, just a few comments

File: MekHQ/data/scenariotemplates/Destroy Grounded Dropship.xml (hunk 2 above)
Comment (NickAragua) [reply MekHQ/data/scenariotemplates/Destroy Grounded Dropship.xml:174]: I think it's a function of how the code is saved, there's probably a "don't serialize empty lists" flag somewhere.

**Pull request merged**